flink-python/setup.py

```diff
@@ -0,0 +1,154 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+from __future__ import print_function
+
+import io
+import os
+import sys
+
+from setuptools import setup
+from shutil import rmtree
+from xml.etree import ElementTree as ET
+
+PACKAGE_NAME = 'apache-flink-connector-aws'
+# Source files, directories
+CURRENT_DIR = os.path.abspath(os.path.dirname(__file__))
+POM_FILE = os.path.join(CURRENT_DIR, '../pom.xml')
+README_FILE = os.path.join(CURRENT_DIR, 'README.txt')
+
+# Generated files and directories
+VERSION_FILE = os.path.join(
+    CURRENT_DIR,
+    'pyflink/datastream/connectors/cassandra_connector_version.py')
+DEPENDENCY_FILE = os.path.join(CURRENT_DIR, 'dev/dev-requirements.txt')
```

Comment:
Don't we need the cassandra connector jars on the classpath?

I expect that they will be removed from the original Flink python release, and we need to add them to the individual packages to make the package working

Reply:
I think I followed the [AWS code](https://github.com/apache/flink-connector-aws/blob/05fdf377c8846cb140ffa3ec96e8ae07282ef4ce/flink-python/setup.py#L72) here, which does not include the JAR.

I am not sure about this. First, going onward the connectors should have separate python releases as well? My first thought is a Python release should not contain the connector JAR. In that case, the JAR will be duplicated if the Java conenctor is used as well. Installing the Java dependencies as a prerequisite for the Python connector to work sounds far to me, but I do not know anything about the Python release model, so I might missing something. :)

Reply:
My feeling is that the connector python release **should** contain the java classes need for it to work.

So the user just installs flink-python, then the connector package and then everything should work. Pushing additional requirements for the user by finding the relevant jar files and installing them manually sounds like a bad move. Discussed this with @gaborgsomogyi, and we found that it is generally a better solution to just include the relevant jar files to the connector package.

Reply:
In short `pip install apache-flink` installs python stuff + jars. Why do we want to do it differently other places?
Having separate jar + python code expects a compat matrix doc which:
* Nobody will do it
* Since nobody knows what is compatible to what ends up in jira issue hell